components/CodeEditor.vue

```diff
@@ -27,20 +27,34 @@ const { data: user } = await useFetch("/api/auth/user");
 const state = reactive({
   code: user.value?.body.code || defaultCode,
   showAPIReference: false,
+  intervalSubmit: false,
 });
 
+const buttonText = ref("Submit");
+
 function onSubmit(event: Event) {
   event.preventDefault();
-
   fetch("/api/bot/submit", {
     method: "POST",
     headers: {
       "Content-Type": "application/json",
     },
     body: JSON.stringify({ code: state.code }),
   });
-}
 
+  state.intervalSubmit = true;
+  let countdown = 4;
+
+  const timer = setInterval(() => {
+    buttonText.value = `${countdown.toFixed(1)}`;
+    countdown -= 0.1;
+    if (countdown < 0) {
+      clearInterval(timer);
+      buttonText.value = "Submit";
+      state.intervalSubmit = false;
+    }
+  }, 100);
```

Comment:
- мы можем здесь считать секундами вместо миллисекунд, т.к. секунды и так достаточно быстро идут
- предлагаю чуть поправить имена переменных для ясности
- предлагаю таймер в 4 секунды заменить на 10 или 30, 4 секунды слишком мало + его можем считывать из какого-нибудь файла с константой, который будет импортировать и фронтенд и бакенд, чтобы эта константа была всегда синхронизирована между и бакендом и фронтендом

```suggestion
  state.submitDisabled = true;
  let countdownSec = 10;
  const timerIntervalSec = 1;

  const timer = setInterval(() => {
    buttonText.value = `${Math.round(countdownSec)}`;
    countdownSec -= timerIntervalSec;
    if (countdownSec < 0) {
      clearInterval(timer);
      buttonText.value = "Submit";
      state.submitDisabled = false;
    }
  }, timerIntervalSec * 1000);
```